backend/games/battle_mogs/src/algorithm.rs

```diff
@@ -438,33 +427,35 @@ impl Generation {
 		input_rarity_1: RarityType,
 		input_generation_2: MogwaiGeneration,
 		input_rarity_2: RarityType,
-		random_hash: &[u8],
+		random_hash: &[u8; 32],
```

Comment:
If we constrain the hash here to the proper length, we can remove the length checks later.